cmd/eval.go

```diff
@@ -205,9 +208,12 @@ Set the output format with the --format flag.
 Schema
 ------
 
-The -s/--schema flag provides a single JSON Schema used to validate references to the input document.
+The -s/--schema flag provides one or more JSON Schemas used to validate references to the input or data documents.
+Loads a single JSON file, applying it to the input document; or all the schema files under the specified directory.
+When a directory is passed, the schema file for input must be named 'default-input-schema.json'.
```

Comment:
😅 Sorry I'm still not completely on board with this design. If it needs that many words to explain, it's perhaps too complicated and will lead to confusion....

Do we have some sort of precedent? Other tools that work the same?

I suppose we could make `--schema` repeatable, and a single file argument would be the one for input, a directory would be a schema set. 🤔 Not sure if that's better...

Reply:
We did contemplate between having `--schema` as a repeatable flag like `--data` and `--bundle`, or only allow for a single path. At the end from our perspective, the latter seems much cleaner and less complicated by reducing the number of scenarios (especially since the path + files a user provides needs to have a simple way of representation in the rule annotation).

Our current design in simple words is just allowing for either of these cases:
1. `opa eval -i input.json -s input-schema.json`.  (don't have any schemas for data docs)
2. `opa eval -i input.json -d datadocs/ -s schemas/ (where schemas directory can contain any or all of the schemas for the input and data docs)

for #2, we impose a rule to have the schema file for input to be named `default-input-schema.json` (to differentiate it from other files (could be data schema files or additional input schema files)) and also means the input annotation doesn't need to be explicitly specified for rules.

Having `--schema` as repeatable means we could have many more complex scenarios, for ex:
`opa eval -i input.json -s input-schema.json -s some-file.json -s schemas/` which isn't clear whether some-file schema should be associated to input or some data doc, and how we reference it in a rule annotation.

Open to further discussion though the current design of limiting the number of schema args feels simpler in our mind :). Wdyt?

Reply:
Thanks for providing some insights into your design tradeoff thoughts 😃 

Let's roll with that, if it turns into a source of UX-related issues, we can revisit. 👍

Reply:
Sounds like a good idea. Thanks Stephan.

Reply:
I apologize in advance for bringing this up again 😅 

I'm hesitant about releasing with the `default-input-schema` convention exposed to users in the CLI as well as annotations because it'll be difficult to unwind. If we drop the convention now, we could always bring it back in the future. It's always much easier to make additive changes than it is to make destructive ones...

Can we drop the `default-input`schema` convention for now? I think this would just amount to (i) removing the special case from the `readSchemaBytes` function below and (ii) removing the special casing that is required

Reply:
@tsandall do you mean that users can either submit a single file, in which case we set the schema for input globally (to maintain backward compatibility with our previous PR), or if they submit a directory then we are not setting the input schema globally at all?

Reply:
@vazirim yes, that's what I had in mind.